src/pkg/packager/create.go

```diff
@@ -66,6 +66,9 @@ func Create(ctx context.Context, packagePath string, output string, opts CreateO
 		if err != nil {
 			return "", fmt.Errorf("failed to load differential package: %w", err)
 		}
+		defer func() {
```

Comment:
observation: does this `defer` offer us any benefit over a direct `pkgLayout.Cleanup()` execution? 

One thought being that - in performing all of the cleanup at the end - we potentially allow the core logic to execute before potentially interrupting with any errors?